### PR TITLE
Single registered article guest view

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #9830
#### Testing Instructions

Create a single article with an intro and full image and a readmore
Set the article to registered only
Create a menu link to the article of type single article
In the menu options make sure show unauthorised links is set to show

Now check the menu item on the front end - you should see the introtext but not the intro image and when you login you see all the text and the full image

Apply the patch
Now check the menu item on the front end - you should see the introtext and the intro image and when you login you see all the text and the full image

### CODE REVIEW

**Comment by Webdongle** [comment]
I have tested this item :red_circle: unsuccessfully on 030de773fdb6dda9ac253d22998f6f23dc8700fb<br /><br />With  Show Voting set to 'Show' ... the voting dropdown box displays twice to guests
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

File: components/com_content/views/article/tmpl/default.php [hunk 1]
```diff
@@ -133,7 +133,14 @@
 	<?php endif; ?>
 	<?php // Optional teaser intro text for guests ?>
 	<?php elseif ($params->get('show_noauth') == true && $user->get('guest')) : ?>
-	<?php echo $this->item->introtext; ?>
+    	<?php echo JLayoutHelper::render('joomla.content.intro_image', $this->item); ?>
```
Comment (smz) [inline components/com_content/views/article/tmpl/default.php]: Move down below the "if"

File: components/com_content/views/article/tmpl/default.php [hunk 2]
```diff
@@ -133,7 +133,14 @@
...
+	<?php if (!$params->get('show_intro')) : ?>
+	<?php // Content is generated by content plugin event "onContentAfterTitle" ?>
+	<?php echo $this->item->event->afterDisplayTitle; ?>
```
Comment (smz) [inline components/com_content/views/article/tmpl/default.php]: already done in line 85 ("voting" is here)
all which is above lie 95 should not be duplicated here

**Comment by smz** [comment]
proposed some changes
sorry for the brevity (on the road...)

**Comment by brianteeman** [comment]
Updated to remove double votes

File: components/com_content/views/article/tmpl/default.php (hunk 1 above)
Comment (smz) [inline components/com_content/views/article/tmpl/default.php]: No, sorry, I was wrong on this: I mistook the if below as the if on "image_intro", but that's already in the JLayout

File: components/com_content/views/article/tmpl/default.php (hunk 2 above)
Comment (smz) [inline components/com_content/views/article/tmpl/default.php]: Also the handling of "afterDisplayTitle" does not belongs here as it has already been performed on lines 87-89 (_will generate duplicate_...)

**Comment by rgmears** [comment]
Works for me.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by woluweb** [comment]
I have tested this item :white_check_mark: successfully on dc144b3b84ceb21afbc23619b18901de82d486bc<br /><br />Tested successfully on [cc44ae1](https://github.com/joomla/joomla-cms/commit/cc44ae16209d260a0750069104bcac759bb37621 "cc44ae1")<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by wojsmol** [comment]
@rgmears Please mark test result on https://issues.joomla.org/tracker/joomla-cms/9865

**Comment by rgmears** [comment]
@wojsmol 
>Please mark test result on ....

I don't know how to do that. Sorry.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by wojsmol** [comment]
@rgmears See https://docs.joomla.org/Testing_Joomla!_patches#Recording_test_results

**Comment by brianteeman** [comment]
@rgmears @wojsmol its ok I manually added his test :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by smz** [comment]
I cannot test now (_on a train with no test environ with me_) but by simple code review it is evident that that lines 137-140 are a duplicate of line 87.

Both echoes the content generated by "afterDisplayTitle" plugins (_probably a very seldom used event, but anyway..._) which in case will be duplicated...

In Joomla event lists (https://docs.joomla.org/Plugin/Events/Content) I don't find "afterDisplayTitle": is it the same as "onContentAfterTitle"?

**Comment by smz** [comment]
In https://docs.joomla.org/Plugin/Events/Content a note in the "onContentAfterTitle" event description says that
> ... this event has special purpose in com_content for use in handling the introtext.

but I'm unsure of what this is or could be as no further specification is given.

What is evident from code is that content generated by this event is displayed only if the "show_intro" param is set.

**Comment by smz** [comment]
Confirmed by testing:

#### Without:
without

#### With:
with

The "Test Aftertitle" plugin is:
~~~
<?php
defined('_JEXEC') or die;

class PlgContentTest_aftertitle extends JPlugin {

	// onContentAfterTitle handler
	public function onContentAfterTitle($context, &$article, &$params, $limitstart) {
		return '<h3>This is generated by the "Test Aftertitle" plugin</h3>';
	}

}
~~~

**Comment by smz** [comment]
I have tested this item :red_circle: unsuccessfully on dc144b3b84ceb21afbc23619b18901de82d486bc<br /><br />See above...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by Webdongle** [comment]
As the patch has been updated because of the fail when tested with the edited plugin ... should the Testing Instructions be updated to take into account that part of the test ?


>In https://docs.joomla.org/Plugin/Events/Content a note in the "onContentAfterTitle" event description says that
>>    ... this event has special purpose in com_content for use in handling the introtext.

Has me totally confused and am unsure how to perform the test with the edited plugin ?

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by smz** [comment]
@Webdongle 
TBH, I'm as much confused as you are, and probably quite more (_but I'll elaborate about that later on_)

As far as regards testing instructions the last word is of course @brianteeman's one, but I can explain what I did in the PR I proposed to Brian and he later incorporated, if this can help.

 1) I eliminated the duplicate display of output generated for the "onContentAfterTitle". This wasn't there at the beginning and was introduced (_by mistake, IMHO_) in Brian's PR. Eliminating this, thus, doesn't change the expected behaviour and hence testing instructions.

 2) I added an extra processing step so that the intro text is passed to the "onContentPrepare" event. This fixes the issue of directives like `{loadposition}` or `{loadmodule}` appearing verbatim in the generated output instead of being processed. I added this as I interpreted this was Brian wish when (_while closing #9830_) he said:

> ... better code than suggested before as it uses the layout correctly **and the plugins (i hope)** (_my emphasis_ )

**This second change requires of course also testing that those directives are indeed honoured** (_as they are, from my testing_).

About this second change I'm also unsure (_and I expressed my doubts directly to Brian_) if it is correct to do that here or would it better to do it here or it would be better to perform it at an higher level (_at the view level, probably_) so that not only this template, but also all other templates using the "intro text" would benefit of the fix.
___________
##### Unsure if off topic (_about the "onContentAfterTitle" event and the "Show Intro Text" option_).

The "Show intro text" option seems to be quite a strange beast. The description is:
> If set to Show, the Intro Text of the article will show when you drill down to the article. If set to Hide, only the part of the article after the "Read More" break will show.

It indeed does that (_intro text not displayed in the "Single Article" view if set to "Hide"_) **but it also controls (_with inverted logic_) weather the content generated by the "onContentAfterTitle" should be displayed or not.**

This is applied, apparently in an inconsistent way, throughout different views. For this, I think we'll need the memory of "_old timers_" to explain why this is done this way and what's the logic behind it (_I'm calling @infograf768 for that, as apparently he did put his hands on that and probably should know..._)

**Comment by conconnl** [comment]
I have tested this item :white_check_mark: successfully on bc7d64c6c2f67d20db97de2483d2767d40d39a79<br /><br />Tested successful per instructions<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by Simonkloostra** [comment]
I have tested this item :white_check_mark: successfully on bc7d64c6c2f67d20db97de2483d2767d40d39a79<br /><br />Testscenario executed successfully, intro-image is now shown in single-article view<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by rgmears** [comment]
I have tested this item :white_check_mark: successfully. 

Intro Image shows. 

`{loadposition` and `{loadmodule}` both display content not code.

**Comment by joris85** [comment]
Tested successfully,
As quest you do not see the intro image while you should see the intro image.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by klatte88** [comment]
Patch tested succesfully after patch intro image shows up as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by klatte88** [comment]
I have tested this item :white_check_mark: successfully on bc7d64c6c2f67d20db97de2483d2767d40d39a79<br /><br />problem as described recreated and solved after patch<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9865">issues.joomla.org/joomla-cms/9865</a>.</sub>

**Comment by rgmears** [comment]
Forgive the uninformed question, but, what does RTC mean?

**Comment by smz** [comment]
@rgmears 
> Forgive the uninformed question, but, what does RTC mean?

**R**eady **T**o **C**ommit

@brianteeman 
So you think it is OK to fix the "onContentPrepare" issue here?

**Comment by rgmears** [comment]
Thanks @smz .

**Comment by brianteeman** [comment]
Create a new issue please

On 15 April 2016 at 15:04, Sergio Manzi <notifications@github.com> wrote:

> @rgmears <https://github.com/rgmears>
>
> Forgive the uninformed question, but, what does RTC mean?
>
> *R*eady *T*o *C*ommit
>
> @brianteeman <https://github.com/brianteeman>
> So you think it is OK to fix the "onContentPrepare" issue here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/9865#issuecomment-210475381>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by smz** [comment]
about what?

**Comment by smz** [comment]
... pinging all who were involved in this as I amended it in #9964: thanks for testing and/or review.

**Pull request merged**